### DISCUSSION TITLE
Multiple fallback when a specific text is not found

### DISCUSSION THREAD

**Discussion by valim2322** [ngneat/transloco#406] [Q&A]
Hi there,
There is a way of fallback multiple times if a specific text is not found in a file?
Assuming that we have fr-CA.json, fr.json and en.json.
For example, I set the the language to fr-CA and I want to show "hello" in fr-CA. If the "hello" entry is not found in fr-CA file it should fallback to fr file. If the text is not found even there, fallback to en (that contains all the text).

Thank you!

**Comment by shaharkazaz** [comment]
@valim2322 Currently transloco doesn't do that out of the box.
The fallback strategy was made for file load failures and not missing keys.
You should avoid these scenarios, what's your use case? you can implement this logic by yourself (if it makes sense to you)

You can use the [transloco keys manager](https://github.com/ngneat/transloco-keys-manager) to help you manage 🙂

**Reply by valim2322** [reply]
Thanks for your reply.
The scenarios is like this: my Angular app will be deployed to multiple clients that can have different translations of the same text. For example part number for client A can be named product number for client B. Besides this, all the client applications needs to be localized. In this case, clientA.fr-CA.json contains only entries that are specific for that client in fr-CA locale. 
With this, I'll end-up with multiple fallbacks, because only the main en.json file contains all the translated text.
For example:
clientA.fr-CA.json fallback to clientA.fr.json, if the text is not found here it fallback to clientA.en.json, if the text is not here it goes to global fr-CA.json and then to global fr.json and finally, it reach the en.json where definitely contains the text.

Thank you!

**Reply by shaharkazaz** [reply]
Thanks for explaining 🙂 

Well, I don't think that this isn't a very common use case and I don't think that currently, this should be part of the main lib.
As I said you can try and implement some logic that suits your needs via the transloco service API.

**Reply by Andrew-Marks-Trisept** [reply]
I'm researching Angular translation solutions and we have the _exact_ same requirements.  I bet a lot of people with multi-site architectures do.  transloco looks great and when I read it supported "multiple fallbacks" I was excited, but if it's only at the file level that is an issue.  I'm actually not sure I understand the use case for fallbacks of missing files in the first place.  Why would anyone configure their app to support a certain language and then not have a translations file for that language?

Anyway, my goal is to have a set of "base" translations for each language with our UX-recommended verbiage, but different clients/sites will still want to customize certain terms.  Similar to valim2322, I wanted to allow entries to be added for specific clients in their files that would "override" the base entries.

@valim2322 did you figure out any solution?

@shaharkazaz when you said "you can try and implement some logic that suits your needs via the transloco service API", is there something more specific you could point me at?  I looked at https://ngneat.github.io/transloco/docs/hack/#fallback-strategy, but it doesn't seem like defining a different FallbackStrategy would help in this case.

**Reply by valim2322** [reply]
I didn't find an out of the box solution. I developed a small C# console app that will be executed during the Angular build that merge multiple non-complete resource json files into a single complete one that later on is used by Angular. The file merging is done by priority, so if a resource key exist in client.fr-CA.json it will overwrite the existing key in global fr-CA.json.

**Reply by Andrew-Marks-Trisept** [reply]
@valim2322 I've never done anything like that, but thanks for the idea!
